WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -183,9 +183,9 @@ def index_for_group_type
 
   def create
     user_id = params.require(:userId)
-    group_id = params.require(:groupId)
+    group_id = params[:groupId] || UserGroup.find_by(group_type: params.require(:groupType)).id
 
-    if group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
+    if group_id.is_a?(String) && group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
```

Comment:
Not a big fan of doing two separate things in one endpoint.

What about something like `def create_by_id` and `create_by_type` instead? I am open to other names if you can think about a better naming.

As a general rule of thumb: When you need to do type-checks in a weakly-typed programming language, there's at least 80% chance that your code design is flawed.

Reply:
Basically, `create` method is to create a new role. The final body of `create` method (at the end of roles project) will be something like:

```
def create
  # initialization of all params like `user_id`, `group_id`, etc.
  # creation of metadata based on `group_type`
  role = UserRole.create!(
    ...
  )
  RoleChangeMailer.notify_role_start(role, current_user).deliver_later
  render json: {
    success: true,
  }
end
```

The current code inside `if` block is to create `team_member`, so that once migrated to `user_roles`, the backend can be silently changed to new way.

Also, I didn't change anything inside `if` block. The `group_id` is ideally integer, but for temporary cases, the group_id is string. So to handle those cases, it's converted into string.

Reply:
Okay, granted as an exception because this will be cleaned up after the transition is complete :)